File: javapatterns/src/main/java/org/iut/nantes/state/car/Attente.java
package org.iut.nantes.state.car;

class Attente extends Etat
{
  public Attente( GoogleCar car )
  {
    super( car );
  }

  public void bouton()
  {
    System.out.println("Bouton appuyé lors de l'état d'attente : on éteint le moteur.");
    car.setEtat( car.getArret() );
  }
  
  public void saisieDestination()
  {
    System.out.println("Destination saisie : en voiture Simone !");
    car.setEtat( car.getConduite() );
    // on conduit pendant une seconde
    try
    {
      Thread.sleep( 1000 );
    }
    catch(Exception e){}
    
    car.getCourant().arrive();
  }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/state/car/etats/Attente.kt
package org.iut.nantes.state.car.etats

import org.iut.nantes.state.car.Car

class Attente(car: Car) : Etat(car) {

    override fun action() {
        println("Voiture en attente d'adresse")
    }

    override fun bouton() {
        println("Extinction du moteur")
        car.changeEtatArret()
    }

    override fun adresse() {
        println("Et c'est parti !")
        car.changeEtatConduite()
        Thread.sleep(1000)
        car.conduire()
    }
}